projects/wave-core-new/src/lib/project/project.service.ts

```diff
@@ -712,6 +717,79 @@ export class ProjectService {
         return ProjectService.subscribeAndProvide(result);
     }
 
+    addDataSetToMap(dataset: DataSet): Observable<void> {
```

Comment:
You should put it into your dataset service. I don't see why it should be in the project service.

Reply:
I put it there first but I was afraid of getting a circular dependency between Project and DatasetService. And adding something to the map might as well be done in ProjectService I think

Reply:
you don't have a dependency from project to datasets

Reply:
6b05d3ccd0295b16ae979a4cca1f1a468a5d2489